pkg/iostreams/iostreams.go

```diff
@@ -135,7 +139,7 @@ func (s *IOStreams) ColorScheme() *ColorScheme {
 	return NewColorScheme(s.ColorEnabled())
 }
 
-func System() *IOStreams {
+func System(configFunc func() (config.Config, error)) *IOStreams {
```

Comment:
So far, IOStreams was pretty application-agnostic. Could we possibly keep that approach and avoid coupling it to Config?

IOStreams can still absolutely react to user's configuration values, but maybe Config and IOStreams should have a facilitator that takes preferences from one and applies them to the other.

Currently, that facilitator can be `main.go`, as it both instantiates `iostreams.System()` (through `factory.New()`) and has access to the user's config at the same time (for expanding aliases and doing an auth check). When `main.go` detects that a user does not wish to see prompts, it could simply invoke something like `IO.SetCanPrompt(false)`. Would that make sense?

Reply:
yep, i agree that would be better